### PR TITLE
[darkforest-score] dark forest custom scoring

### PR DESCRIPTION
**Pull request by cha0sg0d**

### CODE REVIEW

**Comment by cha0sg0d** [comment]
If possible, a quick review would be great. Hoping to use this strategy tomorrow!

File: src/strategies/darkforest-score/index.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  const response = await multicall(
+    network,
+    provider,
+    abi,
+    addresses.map((address: any) => [
+      options.address,
+      'players',
+      [address.toLowerCase()]
+    ]),
```
Comment (ChaituVR) [inline src/strategies/darkforest-score/index.ts:33]: `players` method is taking too long to resolve and failing the tests when we pass more addresses, everything else looks good. make sure it works well and pass all tests when you run

```
npm test --strategy=darkforest-score --more=500
```

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/strategies/darkforest-score/index.ts (hunk 1 above)
Comment (cha0sg0d) [reply src/strategies/darkforest-score/index.ts]: Switched to subgraph for performance, and this passed all tests locally. Thanks!

File: src/strategies/darkforest-score/index.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+// import { getAddress } from '@ethersproject/address';
+import { subgraphRequest } from '../../utils';
+// import { fetch } from "cross-fetch";
```
Comment (ChaituVR) [inline src/strategies/darkforest-score/index.ts:3]:
```suggestion
import { subgraphRequest } from '../../utils';
```

File: src/strategies/darkforest-score/index.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    params.players.__args.block = { number: snapshot };
+  }
+
+  const result = await subgraphRequest(DF_SUBGRAPH_URL, {
+    ...params
+  });
+
+  return Object.fromEntries(
+    result.players.map(p => [
+      p.id,
```
Comment (ChaituVR) [inline src/strategies/darkforest-score/index.ts:47]: @cha0sg0d Here by default thegraph returns lower cased addresses. we need to convert them into checksummed addresses with
```js
import { getAddress } from '@ethersproject/address';
getAddress(p.id);
```

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/strategies/darkforest-score/index.ts (hunk 3 above)
Comment (cha0sg0d) [reply src/strategies/darkforest-score/index.ts]: done!

**Review by bonustrack — APPROVED** [review]

**Review by ChaituVR — APPROVED** [review]

**Pull request merged**